openhands/runtime/base.py

```diff
@@ -677,59 +705,82 @@ def get_microagents_from_org_or_user(
         # Extract the domain and org/user name
         org_name = repo_parts[-2]
 
-        # Construct the org-level .openhands repo path
-        org_openhands_repo = f'{org_name}/.openhands'
-
-        self.log(
-            'info',
-            f'Checking for org-level microagents at {org_openhands_repo}',
-        )
+        # Determine if this is a GitLab repository
+        is_gitlab = self._is_gitlab_repository(selected_repository)
+
+        # For GitLab, try openhands-config first, then .openhands as fallback
+        # For other providers, use .openhands only
+        repo_names_to_try = []
+        if is_gitlab:
+            repo_names_to_try = [
+                f'{org_name}/openhands-config',
+                f'{org_name}/.openhands',
+            ]
+        else:
+            repo_names_to_try = [f'{org_name}/.openhands']
```

Comment:
For gitlab do not try ".openhands" since it is not a legal repo name.
That also means you do not need the for loop below, you can just try the appropriate repo name.